Patch for linux 2

NicBOMB opened the pull request:
This fixes leftovers from the prior https://github.com/Nexus-Mods/node-loot/pull/5.
However, [my comment about the loot library linking error](https://github.com/Nexus-Mods/node-loot/pull/5#issuecomment-1291894377) is still relevant and a fixing commit could be added to this PR before merging.
```
  CXX(target) Release/obj.target/node-loot/src/exceptions.o
  CXX(target) Release/obj.target/node-loot/src/string_cast.o
  CXX(target) Release/obj.target/node-loot/src/napi_helpers.o
  CXX(target) Release/obj.target/node-loot/src/util.o
  SOLINK_MODULE(target) Release/obj.target/node-loot.node
/usr/bin/ld: cannot find -l../loot_api/loot: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [node-loot.target.mk:142: Release/obj.target/node-loot.node] Error 1
```

NicBOMB committed:
remove unused header and value

NicBOMB committed:
make Napi::env args strictly const

This fixes leftovers from the prior #5

NicBOMB committed:
add .cache to gitignore

clang analyzer makes lots of files there.

TanninOne commented:
@NicBOMB I might be wrong but as far as I remember libloot was fairly complex to build - maybe that's limited to windows.
I would rather not rebuild libloot unless we can indeed do that fully automatic through npm on every platform without requiring additional developer tools.

But since libloot binaries are hosted on github, downloading _those_ dynamically for the current platform might be easier.

NicBOMB committed:
fix for linking libloot on Linux

TanninOne committed:
updated libloot to v0.18.3

TanninOne commented:
@NicBOMB The best way to deal with that would probably be to use https://github.com/Nexus-Mods/node-7z
7z can deal with those archives, it exists on all platforms and we need it to work anyway since Vortex uses it to extract mods.

NicBOMB committed:
change loot_api to libloot

Includes an updated build script (libloot.js) and bindings.gyp fixes.

NicBOMB committed:
complete support for Linux

NicBOMB committed:
upgrade dependencies

remove autogypi

NicBOMB committed:
remove autogypi from bindings

NicBOMB committed:
node-gyp is bad

prebuild still depends on an outdated version of node-gyp.
This causes builds to fail in vortex_devel, but not standalone builds.
node-gyp is death by a thousand cuts.
